### PR TITLE
ENH: Enhancements to repo classes 

### PR DESCRIPTION
**Pull request by bpoldrack**
Enhances repository classes in order to refactor PR #52.

### CODE REVIEW

File: datalad/support/annexrepo.py [hunk 1]
```diff
@@ -332,6 +358,144 @@ def annex_add_to_git(self, files):
...
+
+    @normalize_paths
+    def annex_rmurl(self, file, url):
+        """Record that the file is no longer available at the url.
+
+        Parameters:
+        -----------
+        file: str
+
+        url: str
```
Comment (yarikoptic) [inline datalad/support/annexrepo.py]: not sure if that empty line would be "proper"... anyways -- eventually we should not allow a commit go through unless docs build fine (see #116)

File: datalad/tests/test_annexrepo.py [hunk 2]
```diff
@@ -219,3 +234,57 @@ def test_AnnexRepo_annex_add_to_git(src, dst):
...
+
+
+@with_testrepos(flavors=local_flavors)
+@with_tempfile
+def test_AnnexRepo_web_remote(src, dst):
+
+    ar = AnnexRepo(dst, src)
+
+    testurl = 'http://datalad.org/pages/about.html'
+    testfile = 'datalad.org_pages_about.html'
```
Comment (yarikoptic) [inline datalad/tests/test_annexrepo.py]: I still wonder how could we mock external annex... but for this test -- do we really need remote location?  there is ```@serve_path_via_http``` which is often used together with ```@with_tree``` so you could provide a proper http url coming from local box

File: datalad/tests/utils.py [hunk 3]
```diff
@@ -490,6 +492,8 @@ def get_most_obscure_supported_name(tdir):
     TODO: we might want to use it as a function where we would provide tdir
     """
     for filename in OBSCURE_FILENAMES:
+        if on_windows and filename.rstrip() != filename:
```
Comment (yarikoptic) [inline datalad/tests/utils.py]: which reminded me that may be we should have also added upper case letters into the obscure names... added #117 for later

File: datalad/support/annexrepo.py [hunk 4]
```diff
@@ -332,6 +358,144 @@ def annex_add_to_git(self, files):
...
+            options to the annex command
+        """
+
+        annex_options = ['--file=%s' % file[0]] + options + [url]
+        self._run_annex_command('addurl', annex_options=annex_options,
+                                log_online=True, log_stderr=False)
+        # Don't capture stderr, since download progress provided by wget uses
+        # stderr.
+
+    def annex_addurls(self, urls, options=[]):
```
Comment (yarikoptic) [inline datalad/support/annexrepo.py]: later on we will need "geturls" and also we will start thinking about "priorities" so just keep those in mind

Comment (bpoldrack) [inline datalad/support/annexrepo.py]: Yeah - ok.

File: datalad/tests/utils.py (hunk 3 above)
Comment (bpoldrack) [inline datalad/tests/utils.py]: Good point.

**Pull request merged**